ctru-rs/src/applets/error.rs

```diff
@@ -88,17 +88,21 @@ impl PopUp {
     }
 }
 
-/// Sets a custom panic hook that uses the error applet to display panic messages. You can also choose to have
-/// messages printed over stderr along with the pop-up display.
+/// Sets a custom panic hook that uses the error applet to display panic messages. You can also choose to have the
+/// default panic hook called to print the message over stderr.
 ///
-/// If the `Gfx` service is not initialized during a panic, the error applet will not be displayed and the panic
-/// message will be printed over stderr.
-pub fn set_panic_hook(use_stderr: bool) {
+/// If the `Gfx` service is not initialized during a panic, the error applet will not be displayed and the default
+/// hook will be called.
+pub fn set_panic_hook(call_default_hook: bool) {
     use crate::services::gfx::GFX_ACTIVE;
-
-    use std::io::Write;
     use std::sync::TryLockError;
 
+    // Ensure we get the default hook instead of a previously registered user hook.
```

Comment:
Hmm, this is actually the opposite behavior I was thinking, since users might want to use this _then_ their own hook. Since it's just a single call to `take_hook` for the user to restore the default I'd personally prefer this to just say something like `let prev_hook = std::panic::take_hook();`

Reply:
The reason I chose to fall back to the default hook is because there are multiple mechanisms to redirect stdout/stderr on the 3DS (see [Soc::redirect_to_3dslink](https://rust3ds.github.io/ctru-rs/crates/ctru/services/soc/struct.Soc.html#method.redirect_to_3dslink) and [consoleDebugInit](https://libctru.devkitpro.org/console_8h.html#ac4fe073b8a8251a5216b66eef8788da2)), and this implementation makes it easy to still get those printed panic messages over `3dslink` or `GDB` if you want them.

I guess we could allow for calling arbitrary hooks alongside our own, but I'm honestly not convinced how useful that would be compared to ensuring a stable panic hook environment that smoothly hooks into existing APIs that we already support.

Reply:
Changed it to use the previous hook instead of forcing the default